packages/at_client/lib/src/client/at_client_impl.dart

```diff
@@ -214,44 +214,31 @@ class AtClientImpl implements AtClient {
 
   @override
   Future<bool> delete(AtKey atKey, {bool isDedicated = false}) {
-    _telemetry?.controller.sink.add(AtTelemetryEvent('AtClient.delete called', {"key":atKey}));
-    var isPublic = atKey.metadata != null ? atKey.metadata!.isPublic! : false;
-    var isCached = atKey.metadata != null ? atKey.metadata!.isCached : false;
-    var isNamespaceAware =
-        atKey.metadata != null ? atKey.metadata!.namespaceAware : true;
+    _telemetry?.controller.sink.add(AtTelemetryEvent('AtClient.delete called', {"key": atKey}));
     // ignore: no_leading_underscores_for_local_identifiers
-    var _deleteResult = _delete(atKey.key!,
-        sharedWith: atKey.sharedWith,
-        sharedBy: atKey.sharedBy,
-        isPublic: isPublic,
-        isCached: isCached,
-        namespaceAware: isNamespaceAware,
-        isLocal: atKey.isLocal);
-    _telemetry?.controller.sink.add(AtTelemetryEvent('AtClient.delete complete', {"key":atKey, "_deleteResult":_deleteResult}));
+    var _deleteResult = _delete(atKey);
+    _telemetry?.controller.sink.add(AtTelemetryEvent('AtClient.delete complete',{"key":atKey, "_deleteResult": _deleteResult}));
     return _deleteResult;
   }
 
-  Future<bool> _delete(String key,
-      {String? sharedWith,
-      String? sharedBy,
-      bool isPublic = false,
-      bool isCached = false,
-      bool namespaceAware = true,
-      bool isLocal = false}) async {
+  Future<bool> _delete(AtKey atKey) async {
+    // If metadata is null, initialize metadata
+    atKey.metadata ??= Metadata();
     String keyWithNamespace;
-    if (namespaceAware) {
-      keyWithNamespace = _getKeyWithNamespace(key);
+    if (atKey.metadata!.namespaceAware) {
+      keyWithNamespace = AtClientUtil.getKeyWithNameSpace(atKey, _preference!);
```

Comment:
I notice that _getKeyWithNamespace is still called from the `notifyChange` and `putMeta` methods ... do those parts of the code also need to be looked at?

Reply:
`notifyChange` is currently not in use and it has no usage in at_client. The code is moved to `notification_service_impl.notify` method. Perhaps, can we mark it as deprecated? and We have update in `putMeta` method.

Reply:
Yes notifyChange should be marked deprecated

Reply:
I don't understand what you mean by "We have update in putMeta method"?

Reply:
@gkc: Sorry for the typo. I meant it has to be updated in the putMeta method also. Shall I make the changes in the same PR or can I create a separate ticket and do the changes.

Reply:
OK I see. Up to you re separate ticket

Reply:
@gkc: Since this is tested and apps team need the changes, merging these changes. Will make the changes for putMeta in separate PR.